src/PsfFlux.cc

```diff
@@ -94,7 +94,12 @@ void PsfFluxAlgorithm::measure(
         ) {
             badBits |= exposure.getMaskedImage().getMask()->getPlaneBitMask(*i);
         }
-        fitRegion.intersectMask(*exposure.getMaskedImage().getMask(), badBits);
+        auto tmpSpans = afw::geom::SpanSet::fromMask(*exposure.getMaskedImage().getMask(),
+                                                     [&badBits]
+                                                     (afw::image::MaskPixel const & bitPattern)
+                                                     {return bitPattern & badBits;});
+        fitRegion.setSpans(fitRegion.getSpans()->intersectNot(*tmpSpans)->clippedTo(
+            exposure.getMaskedImage().getMask()->getBBox()));
```

Comment:
I'm glad you pointed this out, as it was partly left over from an intermediate development stage. The code now reads:
`fitRegion.setSpans(fitRegion.getSpans()->intersectNot(*exposure.getMaskedImage().getMask(),
            badBits)->clippedTo(exposure.getMaskedImage().getMask()->getBBox()))`
If you ignore the fact that this is sitting inside a footprint, this is exactly the same syntax but now on a spanset object. This SpanSet just must be accessed, and set to, the footprint object, as it itself is immutable. We may consider an api update in the future to add a convenience method to do with from the footprint if it proves necessary. We are hoping that SpanSets will take over in newly developed code such that this sort of thing will be quite rare.